hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/ProtocolMessageSenderProvider.java

```diff
@@ -1,10 +1,18 @@
 package pl.allegro.tech.hermes.consumers.consumer.sender;
 
 import pl.allegro.tech.hermes.api.EndpointAddress;
+import pl.allegro.tech.hermes.api.Subscription;
 
 public interface ProtocolMessageSenderProvider {
 
-    MessageSender create(EndpointAddress endpoint);
+    @Deprecated
+    default MessageSender create(EndpointAddress endpoint) {
+        throw new UnsupportedOperationException();
+    }
+
+    default MessageSender create(Subscription subscription) {
+        return create(subscription.getEndpoint());
```

Comment:
I vote for breaking backwards compatibility and leaving only new one around. Having old one hanging makes it really awkward to implement this interface cause most likely it would be empty.